central/processindicator/datastore/datastore_impl.go

```diff
@@ -148,6 +148,24 @@ func (ds *datastoreImpl) GetProcessIndicator(ctx context.Context, id string) (*s
 	return indicator, true, nil
 }
 
+func (ds *datastoreImpl) GetProcessIndicators(ctx context.Context, ids []string) ([]*storage.ProcessIndicator, bool, error) {
+	indicators, _, err := ds.storage.GetMany(ctx, ids)
+	if err != nil || len(indicators) == 0 {
+		return nil, false, err
+	}
+
+	allowedIndicators := make([]*storage.ProcessIndicator, 0, len(indicators))
```

Comment:
Ah, one more nit then ship. So actually, we don't need to reallocate this because indicators is scoped so we can do a filter trickA

allowedIndicators := indicators[:0]

This will overwrite indicators with the filter and not double allocate